Modules/TPC/src/IDCs.cxx

```diff
@@ -0,0 +1,174 @@
+      for (const auto& key : keyVec) {
+        mStoreMaps.back().insert(std::pair<std::string, std::string>(key, valueVec.at(vecIter)));
+        vecIter++;
+      }
+    }
+    if (keyVec.size() != valueVec.size()) {
+      throw std::runtime_error("Number of keys and values for storeMetaData are not matching");
+    }
+    keyVec.clear();
+    valueVec.clear();
+  }
+
+  for (const auto& entry : config.get_child("qc.postprocessing." + name + ".histogramRanges")) {
+    for (const auto& type : entry.second) {
+      for (const auto& value : type.second) {
+        mRanges[type.first].emplace_back(std::stof(value.second.data()));
+      }
+    }
+  }
+
+  for (const auto& entry : config.get_child("qc.postprocessing." + name + ".timestamps")) {
+    for (const auto& type : entry.second) {
+      mTimestamps[type.first] = std::stol(type.second.data());
+    }
+  }
+
+  mHost = config.get<std::string>("qc.config.conditionDB.url");
+}
+
+void IDCs::initialize(Trigger, framework::ServiceRegistry&)
+{
+  mCdbApi.init(mHost);
+
+  mIDCZeroRadialProf = std::make_unique<TCanvas>("c_sides_IDC0_radialProfile");
+  mIDCZeroStacksA = std::make_unique<TCanvas>("c_GEMStacks_IDC0_1D_ASide");
+  mIDCZeroStacksC = std::make_unique<TCanvas>("c_GEMStacks_IDC0_1D_CSide");
+
+  mIDCDeltaStacksA = std::make_unique<TCanvas>("c_GEMStacks_IDCDelta_1D_ASide");
+  mIDCDeltaStacksC = std::make_unique<TCanvas>("c_GEMStacks_IDCDelta_1D_CSide");
+
+  mIDCOneSides1D = std::make_unique<TCanvas>("c_sides_IDC1_1D");
+
+  mFourierCoeffsA = std::make_unique<TCanvas>("c_FourierCoefficients_1D_ASide");
+  mFourierCoeffsC = std::make_unique<TCanvas>("c_FourierCoefficients_1D_CSide");
+
+  getObjectsManager()->startPublishing(mIDCZeroRadialProf.get());
+  getObjectsManager()->startPublishing(mIDCZeroStacksA.get());
+  getObjectsManager()->startPublishing(mIDCZeroStacksC.get());
+
+  getObjectsManager()->startPublishing(mIDCDeltaStacksA.get());
+  getObjectsManager()->startPublishing(mIDCDeltaStacksC.get());
+
+  getObjectsManager()->startPublishing(mIDCOneSides1D.get());
+
+  getObjectsManager()->startPublishing(mFourierCoeffsA.get());
+  getObjectsManager()->startPublishing(mFourierCoeffsC.get());
+}
+
+void IDCs::update(Trigger t, framework::ServiceRegistry&)
+{
+
+  mCCDBHelper.setIDCZero(mCdbApi.retrieveFromTFileAny<o2::tpc::IDCZero>("TPC/Calib/IDC/IDC0", std::map<std::string, std::string>{}, mTimestamps["IDCZero"])); // fix order: delta before one
+  mCCDBHelper.setIDCOne(mCdbApi.retrieveFromTFileAny<o2::tpc::IDCOne>("TPC/Calib/IDC/IDC1", std::map<std::string, std::string>{}, mTimestamps["IDCOne"]));
+  mCCDBHelper.setIDCDelta(mCdbApi.retrieveFromTFileAny<o2::tpc::IDCDelta<float>>("TPC/Calib/IDC/IDCDELTA", std::map<std::string, std::string>{}, mTimestamps["IDCDelta"]));
+
+  mCCDBHelper.setFourierCoeffs(mCdbApi.retrieveFromTFileAny<o2::tpc::FourierCoeff>("TPC/Calib/IDC/FOURIER", std::map<std::string, std::string>{}, mTimestamps["FourierCoeffs"]));
```

Comment:
Here also the question: Who owns the objects retrieved from the CCDB?

Reply:
Good point. It seems to me that the CdbApi doesn't take care but the mCCDBHelper Object needs to make sure the pointers are deleted. There the pointers to the CCDB objects are members. Including a destructor there to delete them properly should be fine I think. 

So my suggestion would be a destructor in the IDCCCDBHelper for those objects https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/TPC/calibration/include/TPCCalibration/IDCCCDBHelper.h#L145.

Reply:
I think this would not help, since each time you call the `update` function, the pointers to the objects in `mCCDBHelper` will be overwritten and the access to the old ones lost...
I'll make a suggestion.